src/pages/lists.js

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import { DragDropContext as dragDropContext } from 'react-dnd';
+import HTML5Backend from 'react-dnd-html5-backend';
+import PropTypes from 'prop-types';
+import List from '../components/list/';
+import DragSort from '../components/list/dragSort';
+import Button from '../components/button/';
+import Icon from '../components/icon/';
+import Input from '../components/input/';
+
+class Item extends React.Component {
+  state = {
+    textInput: ''
+  };
+
+  handleChange = (e) => {
+    this.setState({ textInput: e.target.value });
+  };
+
+  render() {
+    if (this.props.isEditing) {
+      return (
+        <div className="intellyo-list-item list-item-editing">
+          <Input
+            defaultValue={ this.props.text }
+            onChange={ this.handleChange }
+          />
+          <Button
+            onClick={ () => this.props.onUpdate(this.state.textInput) }
+          >
+            Save
+          </Button>
+        </div>
+      );
+    }
+
+    return (
+      <div className="intellyo-list-item">
+        { this.props.text }
+        { this.props.onRemove &&
+          <Icon
+            onClick={ this.props.onRemove }
+            icon="ion-ios-trash"
+          />
+        }
+      </div>
+    );
+  }
+}
+
+Item.displayName = 'Item';
+
+Item.propTypes = {
+  text: PropTypes.node,
+  onRemove: PropTypes.oneOfType([PropTypes.func, PropTypes.bool]),
```

Comment:
one of func and bool?? how? :D

Reply:
Oh I see now. Since a function is a "truthy" value and part of a condition you don't have to sign it as a boolean type because it's gonna be confusing for others. When you do the following:

```
if (props.iAmAFunction) {
  props.iAmAFunction
}
```

it doesn't mean that the type of the prop `iAmAFunction` is a boolean, you're just checking here whether its value is other then a falsy value like `undefined` or `null`.

A truthy value can be (just to mention the important ones):

a function,
a number which is lower or bigger than 0,
a not empty string,
an array (even if it's empty),
an object (even if it's empty),
true

Falsy values:

`0`,
`''` (empty string),
`false`,
`NaN`,
`null`,
`undefined`

Reply:
I fixed. As we discussed the reason was this unnecessary condition check in `lists.js`:
```
onRemove={ this.state.isEditing && (() => this.removeListItem(card.id)) }
```
which I simply changed to
```
onRemove={ () => this.removeListItem(card.id) }
```